src/KM_Controls.pas

```diff
@@ -121,9 +127,15 @@   TKMControl = class
     procedure SetVisible(aValue: Boolean); virtual;
     procedure SetEnabled(aValue: Boolean); virtual;
     procedure SetAnchors(aValue: TKMAnchorsSet); virtual;
+    function GetSelfAbsLeft: Integer; virtual;
+    function GetSelfAbsTop: Integer; virtual;
+    function GetSelfHeight: Integer; virtual;
+    function GetSelfWidth: Integer; virtual;
   public
     Hitable: Boolean; //Can this control be hit with the cursor?
     Focusable: Boolean; //Can this control have focus (e.g. TKMEdit sets this true)
+
+    IsMarkedToUnfocus: Boolean; //Used for cycle focusing on one TKMPanel

```

Comment:
I don't really like this boolean, can we find a better way? You should be able to use TKMPanel.FocusedControlIndex instead. It looks like FocusedControlIndex is the index of the child which has IsMarkedToUnfocus set?